### PR TITLE
Firewall/v1

### PR DESCRIPTION
**Pull request by victorjulien**
Some initial tests towards https://redmine.openinfosecfoundation.org/issues/7269

### CODE REVIEW

**Comment by catenacyber** [comment]
This is a draft, right ?

**Comment by catenacyber** [comment]
The wording `Some initial tests towards` made me think so.
Are the tests supposed to pass ?

**Comment by jasonish** [comment]
If possible, it might be nice to see what "user" story each scenario is testing?

**Comment by victorjulien** [comment]
Not sure I get it. Each test has a README that describes the test. What else is needed?

**Comment by jasonish** [comment]
> Not sure I get it. Each test has a README that describes the test. What else is needed?

Was thinking of something more applicable to an end-user looking for examples.. Like:
- drop all TLS except with sni www.google.com
- drop all else
Where stuff like 3whs is a lower level implementation detail. Or perhaps that is better suited for the firewall chapter in the docs?

**Comment by victorjulien** [comment]
This is not for users, but are tests for supporting development. The tests are self documenting in my opinion.

**Comment by jufajardini** [comment]
> > Not sure I get it. Each test has a README that describes the test. What else is needed?
> 
> Was thinking of something more applicable to an end-user looking for examples.. Like:
> 
>     * drop all TLS except with sni [www.google.com](http://www.google.com)
> 
>     * drop all else
> 
> 
> Where stuff like 3whs is a lower level implementation detail. Or perhaps that is better suited for the firewall chapter in the docs?

I guess we can add explanations in docs and link to the tests, and then one would complement the other 🤔

File: tests/firewall/firewall-04-tls-sni-enforce/test.rules [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# allow tcp handshake
+
+pass tcp any any <> any 443 (flow:not_established; sid:1;)
+
+# allow tls
+
+pass tls any any <> any 443 (flow:established; tls.sni; content:"raw.githubusercontent.com"; sid:2;)
```
Comment (catenacyber) [inline tests/firewall/firewall-04-tls-sni-enforce/test.rules:7]: And would you like to add a rule like
`pass tls any any <> any 443 (flow:established; tx.progress:<tls.sni; sid:3;)`

right ?

**Review by catenacyber — APPROVED** [review]
Good passing tests for current behavior in my opinion

File: tests/firewall/firewall-04-tls-sni-enforce/test.rules (hunk 1 above)
Comment (catenacyber) [reply tests/firewall/firewall-04-tls-sni-enforce/test.rules:7]: Cool
Do you have a redmine ticket for this improvement ?

SV PR is good like that, and you can make a special SV-only merge
https://github.com/OISF/suricata-verify/pulls?q=is%3Apr+is%3Aopen+review%3Aapproved+label%3A%22tests+pass%22

**Comment by victorjulien** [comment]
Merged in #2139, thanks!

**Pull request closed without merging**